application/hubspot/v0/main.go

```diff
@@ -0,0 +1,478 @@
+//go:generate compogen readme ./config ./README.mdx
+
+package hubspot
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+	"sync"
+
+	hubspot "github.com/belong-inc/go-hubspot"
+	"github.com/instill-ai/component/base"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	taskGetContact          = "TASK_GET_CONTACT"
+	taskCreateContact       = "TASK_CREATE_CONTACT"
+	taskGetDeal             = "TASK_GET_DEAL"
+	taskCreateDeal          = "TASK_CREATE_DEAL"
+	taskGetCompany          = "TASK_GET_COMPANY"
+	taskCreateCompany       = "TASK_CREATE_COMPANY"
+	taskGetTicket           = "TASK_GET_TICKET"
+	taskCreateTicket        = "TASK_CREATE_TICKET"
+	taskGetThread           = "TASK_GET_THREAD"
+	taskRetrieveAssociation = "TASK_RETRIEVE_ASSOCIATION"
+)
+
+var (
+	//go:embed config/definition.json
+	definitionJSON []byte
+	//go:embed config/tasks.json
+	tasksJSON []byte
+	//go:embed config/setup.json
+	setupJSON []byte
+
+	once sync.Once
+	comp *component
+)
+
+type component struct {
+	base.Component
+}
+
+type execution struct {
+	base.ComponentExecution
+	client *CustomClient
+}
+
+// Init returns an implementation of IComponent that implements the greeting
+// task.
+func Init(bc base.Component) *component {
+	once.Do(func() {
+		comp = &component{Component: bc}
+		err := comp.LoadDefinition(definitionJSON, setupJSON, tasksJSON, nil)
+		if err != nil {
+			panic(err)
+		}
+	})
+	return comp
+}
+
+func getToken(setup *structpb.Struct) string {
+	return setup.GetFields()["token"].GetStringValue()
+}
+
+// custom client to support thread task
+func hubspotNewCustomClient(setup *structpb.Struct) *CustomClient {
+	client, err := NewCustomClient(hubspot.SetPrivateAppToken(getToken(setup)))
+
+	if err != nil {
+		panic(err)
+	}
+
+	return client
+}
+
+func (c *component) CreateExecution(sysVars map[string]any, setup *structpb.Struct, task string) (*base.ExecutionWrapper, error) {
+
+	e := &execution{
+		ComponentExecution: base.ComponentExecution{Component: c, SystemVariables: sysVars, Task: task},
+		client:             hubspotNewCustomClient(setup),
+	}
+
+	return &base.ExecutionWrapper{Execution: e}, nil
+}
+
+func (e *execution) Execute(_ context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
```

Comment:
share
I am not super senior in Golang community. So, it is only my preference and share with you.

I currently like to make the main file as clean as possible. [Example](https://github.com/instill-ai/component/blob/main/application/email/v0/main.go#L54)

There are a few benefits.
1. It helps other developers to understand what the components do in high level.
2. It can make dependency clear. [Reason](https://github.com/instill-ai/component/pull/187#discussion_r1662946437)
3. ... more that I cannot write in details because of time limitation.

If you agree on this, would you mind to also follow this pattern?

If you agree, it means 
1. all `input.Fields` will be replaced to `type TaskXXXUserInput`
2. the task implementation will have its own file to make the main clean.

Reply:
Understood, I will rewrite the structure and the codes as well.

Reply:
@chuang8511 I fixed the code structure and followed the format you mention. Let me know if there is any additional change I need to do